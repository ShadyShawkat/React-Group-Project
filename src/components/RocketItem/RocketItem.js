import React from 'react';
import PropTypes from 'prop-types';

const RocketItem = ({ rocket }) => {
  const { name, imageSrc, description } = rocket;

  return (
    <>
      <div className="w-48 h-48 rounded-md overflow-hidden flex flex-shrink-0 flex-grow-0 self-center">
        <img src={imageSrc} alt="rocket" />
      </div>
      <div className="flex flex-col items-start">
        <h3 className="text-xl font-bold">{name}</h3>
        <p>{description}</p>
        <button
          type="button"
          className="bg-sky-600 text-white rounded-sm px-2 py-1 hover:bg-sky-800 mt-auto"
        >
          Reserve Rocket
        </button>
      </div>
    </>
  );
};

RocketItem.propTypes = {
  rocket: PropTypes.shape({
    id: PropTypes.string.isRequired,
    name: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    imageSrc: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
  }).isRequired,
};

export default RocketItem;
