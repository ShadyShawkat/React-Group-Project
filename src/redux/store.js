import { configureStore } from '@reduxjs/toolkit';
import rocketsReducer from './rockets';

const store = configureStore({
  reducer: { rockets: rocketsReducer },
});

export default store;
