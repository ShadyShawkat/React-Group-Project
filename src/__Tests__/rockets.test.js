import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';

import RocketItem from '../components/RocketItem/RocketItem';
import store from '../redux/store';

const rocketsDummyData = [
  {
    rocket_id: '1',
    rocket_name: 'rocket 1',
    rocket_type: 'rocket',
    flickr_images: ['https://via.placeholder.com/150'],
    description: 'Rocket 1',
    reserved: false,
  },
];

let tree = null;
describe('Testing rockets render', () => {
  beforeEach(() => {
    rocketsDummyData.reserved = !rocketsDummyData.reserved;
    tree = render(
      <Provider store={store}>
        <ul className="flex flex-col gap-4 mt-16">
          {rocketsDummyData.map((rocket) => (
            <li
              key={rocket.id}
              className="flex gap-6 bg-white pl-2 pr-5 py-3 rounded-lg"
              data-testid="rocketItem"
            >
              <RocketItem rocket={rocket} />
            </li>
          ))}
        </ul>
      </Provider>,
    );
  });
  it('Tests Rockets.js component render', () => {
    expect(tree).toMatchSnapshot();
    const rocketItems = screen.getAllByTestId('rocketItem');
    expect(rocketItems).toHaveLength(1);
  });
});
