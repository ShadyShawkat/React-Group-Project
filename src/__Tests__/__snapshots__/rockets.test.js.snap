// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing rockets render Tests Rockets.js component render 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <ul
        class="flex flex-col gap-4 mt-16"
      >
        <li
          class="flex gap-6 bg-white pl-2 pr-5 py-3 rounded-lg"
          data-testid="rocketItem"
        >
          <div
            class="md:w-48 md:h-48 w-32 h-32 rounded-md overflow-hidden flex flex-shrink-0 flex-grow-0 self-center relative border-2"
          >
            <img
              alt="rocket"
            />
          </div>
          <div
            class="flex flex-col items-start"
          >
            <h3
              class="text-xl font-bold"
            />
            <p>
              Rocket 1
            </p>
            <button
              class="px-2 py-1 mt-auto rounded-md font-medium bg-sky-600 text-white hover:bg-sky-800"
              data-testid="toggleReservationBtn"
              type="button"
            >
              Reserve Rocket
            </button>
          </div>
        </li>
      </ul>
    </div>
  </body>,
  "container": <div>
    <ul
      class="flex flex-col gap-4 mt-16"
    >
      <li
        class="flex gap-6 bg-white pl-2 pr-5 py-3 rounded-lg"
        data-testid="rocketItem"
      >
        <div
          class="md:w-48 md:h-48 w-32 h-32 rounded-md overflow-hidden flex flex-shrink-0 flex-grow-0 self-center relative border-2"
        >
          <img
            alt="rocket"
          />
        </div>
        <div
          class="flex flex-col items-start"
        >
          <h3
            class="text-xl font-bold"
          />
          <p>
            Rocket 1
          </p>
          <button
            class="px-2 py-1 mt-auto rounded-md font-medium bg-sky-600 text-white hover:bg-sky-800"
            data-testid="toggleReservationBtn"
            type="button"
          >
            Reserve Rocket
          </button>
        </div>
      </li>
    </ul>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
